pegasus/helpers/section_api_helpers.rb

```diff
@@ -602,6 +602,10 @@ def self.update_if_owner(params)
       fields[:script_id] = params[:script][:id].to_i
       DashboardUserScript.assign_script_to_section(fields[:script_id], section_id)
       DashboardUserScript.assign_script_to_user(fields[:script_id], user_id)
+    elsif !params[:course_id] && !params[:script_id]
+      # If a null course (no choice or decide later) is chosen, then update the course and script to be nil
+      fields[:course_id] = nil
+      fields[:script_id] = nil
```

Comment:
Nice find!

It would be great if we also added a test case for this, presumably somewhere over here https://github.com/code-dot-org/code-dot-org/blob/e75cfdf3543646651afe8f7f86bf23bee7024a7c/pegasus/test/test_section_api_helpers.rb#L411